src/gui/qgsadvanceddigitizingdockwidget.h

```diff
@@ -99,6 +98,21 @@ class GUI_EXPORT QgsAdvancedDigitizingDockWidget : public QgsDockWidget, private
           HardLock
         };
 
+        /**
+         * Constraint type
+         * \since QGIS 3.32
+         */
+        enum ConstraintType
```

Comment:
can these be moved to Qgis class, and promoted to enum class please?

Reply:
done